frontend/src/Editor/QueryManager/QueryManager.jsx

```diff
@@ -109,9 +117,27 @@ let QueryManager = class QueryManager extends React.Component {
   };
 
   componentWillReceiveProps(nextProps) {
+    if (this.state.selectedQuery && !nextProps.isQueryPaneDragging && !this.state.paneHeightChanged) {
+      const isQueryChanged = !_.isEqual(
+        this.removeRestKey(this.state.options),
+        this.removeRestKey(this.state.selectedQuery.options)
+      );
+      if (this.state.isFieldsChanged && isQueryChanged) {
+        this.setState({ showSaveConfirmation: true, nextProps });
+        return;
+      } else if (!isQueryChanged && this.state.selectedQuery.kind === 'restapi' && this.state.restArrayValuesChanged) {
+        this.setState({ showSaveConfirmation: true, nextProps });
+        return;
+      }
+    }
     this.setStateFromProps(nextProps);
   }
 
+  removeRestKey = (options) => {
+    delete options.arrayValuesChanged;
```

Comment:
This would modify the object and delete the key in the original object.
Better to copy and remove to compare?